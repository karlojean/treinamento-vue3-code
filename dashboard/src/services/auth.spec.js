import axiosMock from "axios";
import AuthService from "./auth";

jest.mock("axios");

describe("AuthService", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it("should return a token when user login", async () => {
    const token = "123.123.123";

    axiosMock.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: { token } });
    });

    const response = await AuthService(axiosMock).login({
      email: "jean@admin.com",
      password: "123",
    });
    expect(response.data).toHaveProperty("token");
    expect(response).toMatchSnapshot();
  });

  it("should return an user when user register", async () => {
    const user = {
      name: "Jean",
      password: "123",
      email: "jean@admin.com",
    };

    axiosMock.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: user });
    });

    const response = await AuthService(axiosMock).login({
      email: "jean@admin.com",
      password: "123",
    });

    expect(response.data).toHaveProperty("name");
    expect(response.data).toHaveProperty("password");
    expect(response.data).toHaveProperty("email");
    expect(response).toMatchSnapshot();
  });

  it("should throw an error when not found", async () => {
    const errors = { status: 404, statusText: "Not Found" };
    axiosMock.post.mockImplementationOnce(() => {
      return Promise.resolve({ request: errors });
    });

    const response = await AuthService(axiosMock).login({
      email: "igor@igor.me",
      password: "123",
    });
    expect(response.errors).toHaveProperty("status");
    expect(response.errors).toHaveProperty("statusText");
  });
});
