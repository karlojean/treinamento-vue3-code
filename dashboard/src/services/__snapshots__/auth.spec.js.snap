// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthService should return a token when user login 1`] = `
Object {
  "data": Object {
    "token": "123.123.123",
  },
  "errors": null,
}
`;

exports[`AuthService should return an user when user register 1`] = `
Object {
  "data": Object {
    "email": "jean@admin.com",
    "name": "Jean",
    "password": "123",
  },
  "errors": null,
}
`;
